core/client/fs/src/main/java/alluxio/client/file/cache/DefaultMetaStore.java

```diff
@@ -57,6 +61,28 @@ public DefaultMetaStore(CacheEvictor evictor) {
     mEvictor = evictor;
   }
 
+  @Override
+  public boolean hasFile(String fileId) {
+    return mFileMap.containsKey(fileId);
+  }
+
+  @Override
+  public void addFile(String fileId, FileInfo fileInfo) {
+    mFileMap.put(fileId, fileInfo);
+    Metrics.FILES.inc();
+  }
+
+  @Override
+  public void removeFile(String fileId) {
+    mFileMap.remove(fileId);
+    Metrics.FILES.dec();
+  }
+
+  @Override
+  public FileInfo getFile(String fileId) {
+    return Preconditions.checkNotNull(mFileMap.get(fileId));
```

Comment:
Throw PageNotFoundException instead of NPE from here, it encountered this NPE while a remove the local client cache files, and the presto can never resume to work well, only I have to do is restart presto